CMake build issue

I just cloned the repo and tried to build this with no mods. I manually created the "build" dir, and followed the instructions to cd into build, and then run "cmake ..". Here's the output from cmake:


D:\Volpe\Geolocation\Luca_Weihs_range-tree\build>cmake ..
-- Building for: Visual Studio 16 2019
-- The C compiler identification is MSVC 19.25.28612.0
-- The CXX compiler identification is MSVC 19.25.28612.0
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Tools/MSVC/14.25.28610/bin/Hostx64/x64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Tools/MSVC/14.25.28610/bin/Hostx64/x64/cl.exe - works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Tools/MSVC/14.25.28610/bin/Hostx64/x64/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Tools/MSVC/14.25.28610/bin/Hostx64/x64/cl.exe - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - not found
-- Found Threads: TRUE
-- Configuring done
CMake Error at CMakeLists.txt:48 (add_executable):
  No SOURCES given to target: RangeTree_prof


CMake Generate step failed.  Build files cannot be regenerated correctly.


Any thoughts?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,11 +14,11 @@ set (RangeTree_VERSION_MINOR 1)
 find_package(Threads REQUIRED)
 
 if(CMAKE_COMPILER_IS_GNUCXX)
-    add_definitions(-Wall -ansi -Wno-deprecated -pthread)
+    add_definitions(-Wall -ansi -Wno-deprecated)
 endif()
 
 # Flags
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+set(CMAKE_CXX_STANDARD 11)
 
 #-------------------
 # set common include folder for module
@@ -36,26 +36,6 @@ set(EXT_PROJECTS_DIR ${PROJECT_SOURCE_DIR}/ext)
 #add_library(${PROJECT_LIB_NAME} ${SRC_FILES})
 #
 add_subdirectory(${EXT_PROJECTS_DIR}/gtest)
-
-#-------------------
-# Profiling
-#-------------------
-add_definitions(${MSVC_COMPILER_DEFS})
-set(PROFILING_NAME ${PROJECT_NAME_STR}_prof)
-include_directories(${COMMON_INCLUDES})
-
-file(GLOB PROF_SRC_FILES ${PROJECT_SOURCE_DIR}/profiling_test.cpp)
-add_executable(${PROFILING_NAME} ${PROF_SRC_FILES})
-
-#target_link_libraries(${PROFILING_NAME} ${PROJECT_LIB_NAME})
-target_link_libraries(${PROFILING_NAME} ${CMAKE_THREAD_LIBS_INIT})
-
-# including boost
-#set(Boost_INCLUDE_DIR /usr/local/Cellar/boost/1.64.0_1/include)
-#set(Boost_LIBRARY_DIR /usr/local/Cellar/boost/1.64.0_1/lib)
-#find_package(Boost 1.61.0 COMPONENTS system filesystem REQUIRED)
-#include_directories(${Boost_INCLUDE_DIRS})
-#target_link_libraries(${PROFILING_NAME} ${Boost_LIBRARIES})
 
 #-------------------
 # Test

--- a/RangeTree.h
+++ b/RangeTree.h
@@ -43,6 +43,7 @@
 #include <type_traits>
 #include <deque>
 #include <cmath>
+#include <algorithm>
 
 namespace RangeTree {
 

--- a/ext/gtest/CMakeLists.txt
+++ b/ext/gtest/CMakeLists.txt
@@ -28,7 +28,5 @@ set(GTEST_INCLUDE_DIRS ${source_dir}/googletest/include PARENT_SCOPE)
 
 # Specify MainTest's link libraries
 ExternalProject_Get_Property(googletest binary_dir)
-set(GTEST_LIBS_DIR ${binary_dir}/googlemock/gtest PARENT_SCOPE)
-
-
+set(GTEST_LIBS_DIR ${binary_dir}/lib PARENT_SCOPE)
 

